### ISSUE TITLE
Support trigger definitions via CLI with `prefect deployment build`

### ISSUE THREAD

**Issue by serinamarie** [PrefectHQ/prefect#10635]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

Currently, users cannot add trigger policies to deployments

### Describe the proposed behavior

Users should be able to pass in a reference to a trigger policy file or raw json of a trigger. The trigger should be validated on deployment build

### Example Use

`prefect deployment build --trigger trigger.yaml`
`prefect deploy build --trigger '{after:...}'`

**Comment by serinamarie** [comment]
Closing as not planned: for future reference, we specifically will not support this.

**Comment by serinamarie** [comment]
Hi @toro-berlin, I chatted with the team and at present it doesn't sound like we will support this enhancement to an older style of deployments.

**Comment by toro-berlin** [comment]
Thanks for getting back to me and clearing up things so quickly, I appreciate it!

Our internal CI pipeline uses the `prefect deployment build` command. I understand that new features, such as the definition of triggers, will not be supported by this command in the future. Instead, they are already supported by the new way of deploying flows via `prefect deploy`. An enhancement will only be implemented there.

We are currently on a prefect-agent infrastructure and have not yet migrated to worker pools. `prefect deploy` with a `prefect.yaml` would be the preferred way if we have migrated.  `prefect deploy` is not working with agents, as I tested and understand.

For me, it is important to understand if this will also hold for other new Prefect features. With that knowledge, we can act accordingly on our side (migration of our CI pipeline to the new way, etc.).

Thank you!
